tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/descartes/impl/descartes_motion_planner.hpp

```diff
@@ -63,17 +63,25 @@ tesseract_common::StatusCode DescartesMotionPlanner<FloatType>::solve(const Plan
                                                                       PlannerResponse& response,
                                                                       const bool /*verbose*/)
 {
-  if (!problem_generator)
+  std::shared_ptr<DescartesProblem<FloatType>> problem;
+  if (request.data)
+    problem = std::static_pointer_cast<DescartesProblem<FloatType>>(request.data);
```

Comment:
I recommend adding a assert() checking that dynamic cast does not equal nullptr.